test/modules/generic-mqtt-tester/src/tester.rs

```diff
@@ -293,22 +293,30 @@ impl MessageTester {
 fn message_handler(
     settings: &Settings,
     publish_handle: PublishHandle,
-) -> Option<Box<dyn MessageHandler + Send>> {
-    let tracking_id = settings.tracking_id();
+) -> Result<Option<Box<dyn MessageHandler + Send>>, MessageTesterError> {
     let relay_topic = settings.relay_topic();
     let module_name = settings.module_name();
+    let batch_id = settings.batch_id();
     let test_result_coordinator_url = settings.trc_url();
     let reporting_client = TrcClient::new(test_result_coordinator_url);
 
     match settings.test_scenario() {
-        TestScenario::InitiateAndReceiveRelayed | TestScenario::Receive => Some(Box::new(
-            ReportResultMessageHandler::new(reporting_client, tracking_id, &module_name),
-        )),
-        TestScenario::Relay => Some(Box::new(RelayingMessageHandler::new(
+        TestScenario::InitiateAndReceiveRelayed | TestScenario::Receive => {
```

Comment:
I see - so only some scenarios need the tracking_id so you made it an Option type so that it doesn't have be specified in the config/settings file for tests that don't need it?

Reply:
yep